compiler/rustc_trait_selection/src/traits/wf.rs

```diff
@@ -792,13 +792,8 @@ impl<'a, 'tcx> TypeVisitor<TyCtxt<'tcx>> for WfPredicates<'a, 'tcx> {
             }
 
             ty::Alias(ty::Opaque, ty::AliasTy { def_id, args, .. }) => {
-                // All of the requirements on type parameters
-                // have already been checked for `impl Trait` in
-                // return position. We do need to check type-alias-impl-trait though.
-                if self.tcx().is_type_alias_impl_trait(def_id) {
-                    let obligations = self.nominal_obligations(def_id, args);
-                    self.out.extend(obligations);
-                }
+                let obligations = self.nominal_obligations(def_id, args);
+                self.out.extend(obligations);
```

Comment:
very much in favor of this :grin: wanted to actually open a PR for that myself soon.

Can you merge the behavior of all `ty::Alias` here?